Since musl 1.2 time_t is a 64 bit value, even on 32 bit systems. A
hotfix for printing the value is simply using PRId64 from inttypes.h
in the format string.

See: https://github.com/jonas/tig/issues/1084

diff -upr tig-2.5.2.orig/src/util.c tig-2.5.2/src/util.c
--- tig-2.5.2.orig/src/util.c	2021-02-14 13:03:52.467947432 +0100
+++ tig-2.5.2/src/util.c	2021-02-14 13:04:22.911283635 +0100
@@ -11,6 +11,8 @@
  * GNU General Public License for more details.
  */
 
+#include <inttypes.h>
+
 #include "tig/tig.h"
 #include "tig/util.h"
 
@@ -160,12 +162,12 @@ get_relative_date(const struct time *tim
 
 		seconds /= reldate[i].in_seconds;
 		if (compact) {
-			if (!string_nformat(buf, buflen, NULL, "%s%ld%c",
+			if (!string_nformat(buf, buflen, NULL, "%s%"PRId64"%c",
 				    now.tv_sec >= timestamp ? "" : "-",
 				    seconds, reldate[i].compact_symbol))
 				return "";
 
-		} else if (!string_nformat(buf, buflen, NULL, "%ld %s%s %s",
+		} else if (!string_nformat(buf, buflen, NULL, "%"PRId64" %s%s %s",
 				    seconds, reldate[i].name,
 				    seconds > 1 ? "s" : "",
 				    now.tv_sec >= timestamp ? "ago" : "ahead"))
